ecs/client.go

```diff
@@ -162,14 +162,45 @@ func (c *BasicECSClient) RunTask(ctx context.Context, in *ecs.RunTaskInput) (*ec
 		out, err = c.ecs.RunTaskWithContext(ctx, in)
 		if awsErr, ok := err.(awserr.Error); ok {
 			grip.Debug(message.WrapError(awsErr, msg))
+			if strings.Contains(awsErr.Error(), "provisioning capacity limit exceeded") {
```

Comment:
For future sustainability can we check if this `awserr` is a specific exception that  implements the interface rather than a hard coded error message substring? I'm concerned this exact substring might change as time goes on.

Reply:
The type of this is a somewhat generic [`ClientException`](https://docs.aws.amazon.com/sdk-for-go/api/service/ecs/#ClientException), of which many client exceptions are genuine client-side errors. In this case, the issue is actually on ECS's side since it needs to scale up.

I also don't really like having to check the exact substring, but I don't think there's a more elegant way to check for the provisioning limit error. If it's any consolation, I've never seen an AWS error message get changed and even if it did change to a different message, it would at least show up in our error alert.

Reply:
Fair enough, SGTM. Glad we have the alerts up in case.